### DISCUSSION TITLE
How to avoid flycheck to mark outdated positions in Go code

### DISCUSSION THREAD

**Discussion by bronger** [emacs-lsp/lsp-mode#3887] [Q&A]
I’m not sure whether I hit a bug, so I first ask for help. I use lsp-mode and flycheck with Go. Because I want to avoid false markings of syntax errors while I type, I set `(setq flycheck-check-syntax-automatically '(save))`.

However, when I save and goimports adds import lines, almost all line numbers change and this confuses flymake. It marks errors in the _old_ lines, which leads to garbled flycheck indicators (red underlinings in my setup).

What can I do about it?

I also tried flymake, however, I failed to find a counterpart of `(setq flycheck-check-syntax-automatically '(save))` in the flymake world. It perpetually checks for errors after each keystroke.

**Comment by yyoncho** [comment]
Try setting `lsp-diagnostic-clean-after-change` to `t`?

**Reply by bronger** [reply]
This means that I have to save twice before an error gets marked.

**Reply by yyoncho** [reply]
Can you attach the client/server communication (preferably as simple as possible file and containing as little as possible information). To me, it sounds like we are ignoring the correct diagnostics.

**Reply by bronger** [reply]
I now prepared a quite minimal example and will file an issue proper.
